standalone-scripts/locustfile_download_upload.py

```diff
@@ -0,0 +1,623 @@
+from locust.runners import MasterRunner, LocalRunner
+
+r = random.Random()
+
+survey_short_name = 'ASHE'
+survey_long_name = 'Annual Survey of Hours and Earnings'
+survey_ref = '141'
+form_type = '0001'
+eq_id = '2'
+period = '1806'
+respondents = int(os.getenv('test_respondents'))
+logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+logger = logging.getLogger()
+
+auth = (os.getenv('security_user_name'), os.getenv('security_user_password'))
+
+# Ignore these during collection exercise event processing as they are the key
+# for the collection exercise and don't represent event data
+ignore_columns = ['surveyRef', 'exerciseRef']
+CSRF_REGEX = re.compile(r'<input id="csrf_token" name="csrf_token" type="hidden" value="(.+?)"\/?>')
+# USER_WAIT_TIME_WAIT_TIME is between GET and POST requests
+USER_WAIT_TIME_MIN_SECONDS = int(os.getenv('user_wait_time_min_seconds', 5))
+USER_WAIT_TIME_MAX_SECONDS = int(os.getenv('user_wait_time_max_seconds', 15))
+
+
+# Load data for tests
+def load_data():
+    logger.info("Container host: %s", socket.gethostname())
+
+    survey_id = load_ashe_survey(auth)
+    load_collection_exercises(auth)
+    load_collection_exercise_events(auth)
+    load_and_link_collection_instrument(auth, survey_id)
+    load_and_link_sample(auth)
+    execute_collection_exercise(auth, survey_id)
+    register_users(auth)
+
+
+# Survey loading
+def load_ashe_survey(auth):
+    logger.info('Trying to find survey %s', survey_short_name)
+    get_url = f"{os.getenv('survey')}/surveys/shortname/{survey_short_name}"
+    get_response = requests.get(get_url, auth=auth)
+
+    try:
+        get_response.raise_for_status()
+        get_data = get_response.json()
+        logger.info("Survey successfully found at id %s", get_data['id'])
+        return get_data['id']
+    except requests.exceptions.HTTPError:
+        logger.error("Couldn't find survey %s, status code %s, message %s", survey_short_name, get_response.status_code,
+                     get_response.text)
+
+    create_url = f"{os.getenv('survey')}/surveys"
+    survey_details = {"surveyRef": survey_ref, "longName": survey_long_name, "shortName": survey_short_name,
+                      "legalBasisRef": 'STA1947', "surveyType": 'Business',
+                      "classifiers": [{"name": "COLLECTION_INSTRUMENT", "classifierTypes": ["FORM_TYPE"]},
+                                      {"name": "COMMUNICATION_TEMPLATE", "classifierTypes": ["LEGAL_BASIS", "REGION"]}]}
+
+    create_response = requests.post(create_url, json=survey_details, auth=auth)
+    try:
+        create_response.raise_for_status()
+        create_data = json.loads(create_response.text)
+        logger.info("Successfully created survey at id %s", create_data['id'])
+        return create_data['id']
+    except requests.exceptions.HTTPError:
+        if create_response.status_code == 409:
+            # it exists try to retrieve it again
+            return load_ashe_survey(auth)
```

Comment:
I can't imagine a situation where someone messes up survey so badly that it returns 409s when its not meant to, but it's probably worth adding an attempt counter in here just to ensure we don't infinitely loop